core/src/main/java/org/apache/accumulo/core/fate/AbstractFateStore.java

```diff
@@ -275,6 +269,10 @@ protected void verifyFateKey(FateId fateId, Optional<FateKey> fateKeySeen,
         "Collision detected for fate id " + fateId);
   }
 
+  protected void verifyLock(ZooUtil.LockID lockID, FateId fateId) {
```

Comment:
Yeah, it is redundant but, like you said, I thought it might be better to have the more detailed error message at the cost of double-checking the lock id for null